### PR TITLE
Update env for upi vsphere zone config

### PR DESCRIPTION
**Pull request by WenXinWei**
update the hardcode master/worker replicas with dynamic env for upi-conf-vsphere-zones-command.sh to unblock https://github.com/openshift/installer/pull/6999

### CODE REVIEW

**Comment by WenXinWei** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-vsphere-upi-zones

**Comment by WenXinWei** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-vsphere-upi-zones

**Comment by WenXinWei** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-vsphere-upi-zones

**Comment by WenXinWei** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-vsphere-upi-zones

**Comment by WenXinWei** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-vsphere-upi-zones

**Comment by WenXinWei** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-vsphere-upi-zones

**Comment by WenXinWei** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-vsphere-upi-zones

**Comment by WenXinWei** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-vsphere-upi-zones

**Comment by WenXinWei** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-vsphere-upi-zones

**Comment by WenXinWei** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-vsphere-upi-zones

**Comment by WenXinWei** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-vsphere-upi-zones

**Comment by WenXinWei** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-vsphere-upi-zones

**Comment by WenXinWei** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-vsphere-upi-zones

**Comment by WenXinWei** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-vsphere-upi-zones

**Comment by WenXinWei** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-vsphere-upi-zones

**Comment by WenXinWei** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-vsphere-upi-zones

**Comment by WenXinWei** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-vsphere-upi-zones

**Comment by WenXinWei** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-vsphere-upi-zones

**Comment by WenXinWei** [comment]
@jianlinliu @jinyunma  please help review the pr when you have time, thanks.

**Comment by WenXinWei** [comment]
> What is the motive to update the logic? We are going to use the replica ENV to do some overriding?

for the PR:https://github.com/openshift/installer/pull/6999, the job 'e2e-vsphere-upi-zones' failed with :
   │ local.compute_fqdns is tuple with 3 elements
    │ module.ipam_compute.ip_addresses is list of string with 4 elements.
    
  in variables.tf, the compute_count default count is 3. The number of compute_ip_address before are 4 -->["192.168.${third_octet}.7", "192.168.${third_octet}.8","192.168.${third_octet}.9", "192.168.${third_octet}.10'']. there will be conflicts. So need to changed the hard-code compute_ip_addresses to  dynamic to suit the WORKER_REPLICAS (4)set in e2e-vsphere-upi-zones.

**Comment by WenXinWei** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-vsphere-upi-zones

File: ci-operator/step-registry/upi/conf/vsphere/zones/upi-conf-vsphere-zones-commands.sh [hunk 1]
```diff
@@ -186,6 +186,25 @@ networking:
...
+num=$((6 +  $WORKER_REPLICAS))
+for i in $(seq 7 $num); do
+    worker_ip="192.168.${third_octet}.$i"
+    echo $worker_ip
+    zone_worker_ip+="\"$worker_ip\"",
+done
+
+declare zone_master_ip
+
+num=$((3 +  $MASRER_REPLICAS))
```
Comment (jinyunma) [inline ci-operator/step-registry/upi/conf/vsphere/zones/upi-conf-vsphere-zones-commands.sh:200]: when $MASRER_REPLICAS > 3, the ip will overlap with worker's ip. The worker `num`  should be based on master `num`.

**Comment by WenXinWei** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-vsphere-upi-zones

**Comment by WenXinWei** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-vsphere-upi-zones

File: ci-operator/step-registry/upi/conf/vsphere/zones/upi-conf-vsphere-zones-commands.sh [hunk 2]
```diff
@@ -186,6 +186,28 @@ networking:
 EOF
 fi
 
+if [ ${MASTER_REPLICAS} -ne 0 ] && [ ${WORKER_REPLICAS} -ne 0 ]; then
```
Comment (jianlinliu) [inline ci-operator/step-registry/upi/conf/vsphere/zones/upi-conf-vsphere-zones-commands.sh:189]: 1. if use `if` block here, pls add the leading white spaces for the following lines.
2. Suggest to use two `if` blocks to judge `MASTER_REPLICAS` and `WORKER_REPLICAS` case separately.

**Comment by WenXinWei** [comment]
/pj-rehearse pull-ci-openshift-installer-master-e2e-vsphere-upi-zones

File: ci-operator/step-registry/upi/conf/vsphere/zones/upi-conf-vsphere-zones-commands.sh (hunk 2 above)
Comment (WenXinWei) [reply ci-operator/step-registry/upi/conf/vsphere/zones/upi-conf-vsphere-zones-commands.sh]: thanks for advice, only worker nums can be set to 0 when compact cluster scenario, so finally i add a if block for worker part.

**Comment by jinyunma** [comment]
@jcpowermac @rvanderp3 could you help to review? thanks.

**Pull request merged**